src/main/java/com/semmle/jira/addon/workflow/LgtmIssueLinkFunction.java

```diff
@@ -0,0 +1,90 @@
+package com.semmle.jira.addon.workflow;
+
+import com.atlassian.jira.bc.issue.link.RemoteIssueLinkService;
+import com.atlassian.jira.bc.issue.link.RemoteIssueLinkService.CreateValidationResult;
+import com.atlassian.jira.bc.issue.link.RemoteIssueLinkService.RemoteIssueLinkResult;
+import com.atlassian.jira.component.ComponentAccessor;
+import com.atlassian.jira.issue.Issue;
+import com.atlassian.jira.issue.MutableIssue;
+import com.atlassian.jira.issue.link.RemoteIssueLink;
+import com.atlassian.jira.issue.link.RemoteIssueLinkBuilder;
+import com.atlassian.jira.user.ApplicationUser;
+import com.atlassian.jira.workflow.function.issue.AbstractJiraFunctionProvider;
+import com.opensymphony.module.propertyset.PropertySet;
+import com.opensymphony.workflow.WorkflowException;
+import com.semmle.jira.addon.Request;
+import com.semmle.jira.addon.Request.Transition;
+import com.semmle.jira.addon.util.Constants;
+import com.semmle.jira.addon.util.Util;
+import java.io.IOException;
+import java.util.Map;
+import org.codehaus.jackson.JsonNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LgtmIssueLinkFunction extends AbstractJiraFunctionProvider {
+
+  private static final Logger log = LoggerFactory.getLogger(LgtmIssueLinkFunction.class);
+
+  @Override
+  @SuppressWarnings("rawtypes")
+  public void execute(Map transientVars, Map args, PropertySet ps) throws WorkflowException {
+
+    if (!ComponentAccessor.getIssueLinkManager().isLinkingEnabled()) {
+      log.debug(
+          "LgtmIssueLinkFunction: Remote issue link creation skipped, because issue linking is disabled.");
```

Comment:
What happens if it is disabled and we create the links, they are just not visible? If later the issue linking is re enabled this links will appear? If so, I think we should create it always, but log the message just in case someone doesn't know whats is happening.

Reply:
No, the IssueLinkService will throw an exception, if called when issue linking is disabled.